### ISSUE TITLE
Duplicated Figure/Table labels in word_document2

### ISSUE THREAD

**Issue by dewittpe** [rstudio/bookdown#1223]
I have taken the example .Rmd file from the [cross-ref](https://bookdown.org/yihui/rmarkdown-cookbook/cross-ref.html) 
documentation and added `bookdown::word_document2: default` in the yaml header.   

    ---
    title: Cross-referencing figures, tables, and equations
    author: Generated by bookdown
    output:
      bookdown::word_document2: default
      bookdown::pdf_document2: default
      bookdown::html_document2: default
    ---
    
    See Figure \@ref(fig:cars-plot).
    
    ```{r cars-plot, fig.cap="The cars data.", echo=FALSE}
    par(mar = c(4, 4, .2, .1))
    plot(cars)  # a scatterplot
    ```
    
    Also see Equation \@ref(eq:mean).
    
    \begin{equation}
    \bar{X} = \frac{\sum_{i=1}^n X_i}{n} (\#eq:mean)
    \end{equation}
    
    And see Table \@ref(tab:mtcars).
    
    ```{r mtcars, echo=FALSE}
    knitr::kable(mtcars[1:5, 1:5], caption = "The mtcars data.")
    ```

The example renders as expected when generating a pdf, but, when using the dropdown options in RStudio to knit to `word_document2` the resulting `.docx` file has the labels for the figure and table duplicated.

Expected "Figure 1: The cars data", observed: "Figure 1: Figure 1: The cars data."

![image](https://user-images.githubusercontent.com/3136617/128569605-95bc4198-7b33-4805-a19a-1ae989ea5ad9.png)

A similar issue for the Table.

```r
> rmarkdown::pandoc_version()
[1] ‘2.14.1’
> xfun::session_info("bookdown")
R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 11.3.1, RStudio 1.4.1717

Locale: en_US.UTF-8 / en_US.UTF-8 / en_US.UTF-8 / C / en_US.UTF-8 / en_US.UTF-8

Package version:
  base64enc_0.1.3   bookdown_0.22.17  digest_0.6.27     evaluate_0.14     glue_1.4.2        graphics_4.1.0   
  grDevices_4.1.0   highr_0.9         htmltools_0.5.1.1 jquerylib_0.1.4   jsonlite_1.7.2    knitr_1.33       
  magrittr_2.0.1    markdown_1.1      methods_4.1.0     mime_0.11         rlang_0.4.11      rmarkdown_2.10   
  stats_4.1.0       stringi_1.7.3     stringr_1.4.0     tinytex_0.33      tools_4.1.0       utils_4.1.0      
  xfun_0.25         yaml_2.2.1        
```

**Comment by yihui** [comment]
This is due to a change in Pandoc 2.14.1: https://github.com/jgm/pandoc/releases/tag/2.14.1 Figures and tables with captions are automatically numbered by Pandoc now. Unfortunately I don't see an option to disable that behavior.

We could disable our numbering in bookdown, but the tricky thing is that we need to make sure the cross-reference numbers are correctly generated: when sections are numbered, bookdown will number figures and tables by X.Y where X is the section number and Y is an incremental number within the section. With Pandoc 2.14.1, we have to disable the numbering scheme X.Y but use a sequential numbering scheme throughout the document.

I'll see what I can do. Thanks for the report!

**Comment by yihui** [comment]
@dewittpe Should be fixed now. You can install the development version via

```r
remotes::install_github('rstudio/bookdown')
```

---

@cderv I didn't investigate deeply on Pandoc's side, and I don't really like the fix #1229. If possible, I wish Pandoc could provide a way to disable its automatic numbering of figures and tables (and ideally, still preserving the capability of generating List of Figures and Tables in Word).

The main drawbacks of #1229 are:

1. Figures can't be numbered by sections in Word;
2. There is no way to customize the figure labels (Pandoc hard-coded the label "Figure").

You may take a look when you are back. Thanks!

**Comment by dewittpe** [comment]
> @dewittpe Should be fixed now. You can install the development version via
> 
> ```r
> remotes::install_github('rstudio/bookdown')
> ```

@yihui - Thank you!

**Comment by cderv** [comment]
Thank @yihui for the quick fix. 

The issue seems now to be fixed on Pandoc's side. (Reported in https://github.com/jgm/pandoc/issues/7499, fixed in https://github.com/jgm/pandoc/commit/4340bd52c49b987087e123de2eae4464b300332f)

Pandoc 2.14.1 will be the only version with the issue I believe, as of now it will work this way: 
* Pandoc's figure numbering for docx is disabled by default
* It can be activated with adding extension `+native_numbering` to the format

However, I am not sure about the LOF and LOT still working. (This would maybe require support on our side by inserting the correct rawInline openxml as they did)

RStudio IDE have rolled back to 2.14.0.3 so it should not be an issue anymore in the daily version I think. 

Should we keep your fix only for Pandoc 2.14.1 or should we just _ignore_ this version by warning to use a different Pandoc version if we detect this version + docx in bookdown (and revert #1229 at next Pandoc release) ?

**Comment by yihui** [comment]
I'll revert #1229 and just ignore this particular version. I guess it shouldn't take very long for the next version of Pandoc to be released, so hopefully this won't affect too many people. If it does affect anyone in the future, they have two options:

1. `remotes::install_version('bookdown', '0.23')` (this is only necessary before **bookdown** 0.24 is released on CRAN, which won't happen very soon);
2. Or install RStudio daily, or wait for the next version of Pandoc to be released and bundled in RStudio daily.
